IdentityCore/src/requests/sdk/MSIDTokenResponseValidator.m

```diff
@@ -64,6 +65,16 @@ - (MSIDTokenResult *)validateTokenResponse:(MSIDTokenResponse *)tokenResponse
         return nil;
     }
     
+    // In case MSAL requests AT POP but Broker responds with AT Bearer, MSAL returns error to user
+    NSString *tokenType = [tokenResponse.tokenType lowercaseString];
+    NSString *tokenTypeFromConfiguration = [configuration.authScheme.tokenType lowercaseString];
+    
+    if (![NSString msidIsStringNilOrBlank:tokenType] && ![tokenType isEqualToString:tokenTypeFromConfiguration])
+    {
+        MSIDFillAndLogError(error, MSIDErrorServerInvalidResponse, @"Please update Microsoft Authenticator to the latest version. Pop tokens are not supported with this broker version.", correlationID);
```

Comment:
This can also happen if server returns unexpected token type. Can we make the message more generic?